presto-main/src/main/java/com/facebook/presto/operator/InMemoryJoinHash.java

```diff
@@ -170,7 +183,7 @@ private boolean positionEqualsCurrentRow(int leftPosition, int rightPosition, Pa
         int blockIndex = decodeSliceIndex(pageAddress);
         int blockPosition = decodePosition(pageAddress);
 
-        return pagesHashStrategy.positionEqualsRow(blockIndex, blockPosition, rightPosition, rightPage);
+        return pagesHashStrategy.positionEqualsRowIgnoreNulls(blockIndex, blockPosition, rightPosition, rightPage);
```

Comment:
This doesn't look correct.  This method is named `positionEqualsCurrentRow` and it is calling `positionEqualsRowIgnoreNulls`, so at the very least the method is misnamed.  But, looking at the usages of this method in `WindowOperator`, the code doesn't seem to be handling the null case, and since types are not expected to handle the  null case, you can get incorrect answers.

I expect that you will need both the `positionEqualsRow` and `positionEqualsRowIgnoreNulls` methods.